### PR TITLE
Password change notification

### PR DESCRIPTION
**Pull request by simeonbizhub**
[Trello](https://trello.com/c/QZ1zbZLa)

Following every password change, an email is now sent to the user confirming their password reset including details date/time, IP address, region (if GeoIP enabled). Additionally, a security log entry is created following every password change.

**Changes**
- Added `ChangePasswordNotificationJob`
- Set `lastChangedDateTime`, `lastChangedIPAddress`, `lastChangedRegion` in `UserBizlet.preSave`
- Send email notification & add security log entry in `UserBizlet.postSave`

### CODE REVIEW

File: skyve-war/src/main/java/modules/admin/ChangePassword/ChangePasswordNotificationJob.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * @author Simeon Solomou
+ */
+public class ChangePasswordNotificationJob extends Job {
+
+	private static final String EMAIL_SUBJECT = "Security alert: password change";
+	private static final String EMAIL_BODY = "Hello %s,"
+			+ "<br/><br/>"
+			+ "This is a courtesy email to let you know that your password has been changed as requested."
+			+ "<br/><br/>"
+			+ "If you have any concerns regarding this, please contact us <a href=\"mailto: info@bizhub.com.au\">here</a>.";
```
Comment (benpetito) [inline skyve-war/src/main/java/modules/admin/ChangePassword/ChangePasswordNotificationJob.java]: This should not be hard coded. Biz Hub do not want to be notified if someone else is responsible for the application. Use `Util.getSupportEmailAddress()` instead.

File: skyve-war/src/main/java/modules/admin/ChangePassword/ChangePasswordNotificationJob.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+			return;
+		}
+		setPercentComplete(25);
+
+		try {
+			EXT.sendMail(new Mail()
+					.addTo(email)
+					.from(UtilImpl.SMTP_SENDER)
+					.subject(EMAIL_SUBJECT)
+					.body(String.format(EMAIL_BODY, contact.getName())));
```
Comment (benpetito) [inline skyve-war/src/main/java/modules/admin/ChangePassword/ChangePasswordNotificationJob.java]: Can this use `Communication` instead? That way, the subject and body text of the email can be customised within the application if necessary.

File: skyve-war/src/main/java/modules/admin/ChangePassword/ChangePasswordNotificationJob.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ * This job sends a notification email to the current user, confirming that they just changed their password.
+ * 
+ * @author Simeon Solomou
+ */
+public class ChangePasswordNotificationJob extends Job {
+
+	private static final String EMAIL_SUBJECT = "Security alert: password change";
+	private static final String EMAIL_BODY = "Hello %s,"
+			+ "<br/><br/>"
+			+ "This is a courtesy email to let you know that your password has been changed as requested."
```
Comment (benpetito) [inline skyve-war/src/main/java/modules/admin/ChangePassword/ChangePasswordNotificationJob.java]: It may not have been requested. I don't have a password change email handy, but maybe reword this slightly. You know the "If you just changed your password, please ignore this".

**Review by benpetito — CHANGES_REQUESTED** [review]

**Comment by benpetito** [comment]
@simeonbizhub , I think we should also send a notification when it is changed from the UI in the admin module. There should not be a way to change a user's password without notifying them.

**Comment by simeonbizhub** [comment]
> @simeonbizhub , I think we should also send a notification when it is changed from the UI in the admin module. There should not be a way to change a user's password without notifying them.

Thanks @benpetito, change requests resolved and this is now achieved.

**Comment by simeonbizhub** [comment]
Awaiting merge of [GeoIP implementation](https://github.com/skyvers/skyve/pull/125), so that geolocation can be included in email contents.

**Comment by simeonbizhub** [comment]
@benpetito Implementation has been updated to include region from the GeoIP implementation. I have updated the details in the PR description. Thanks.
